src/ephemeris.c

```diff
@@ -250,12 +248,28 @@ s8 calc_sat_state(const ephemeris_t *e, const gps_time_t *t,
  */
 u8 ephemeris_valid(const ephemeris_t *eph, const gps_time_t *t)
 {
+  assert(eph != NULL);
+  assert(t != NULL);
+  assert(eph->fit_interval > 0);
```

Comment:
I don't think this should assert on fit interval. Maybe just warn if everything else is ok. I believe all fields zero with valid = false needs to be tolerated (but obviously return false).